frontend/src/reducers/Reducer.ts

```diff
@@ -0,0 +1,31 @@
+import setActions from './StateReducers';
+import setState from './ActionReducers';
+import { State, Action } from '../constants/Interfaces';
+import { HeaderStates, BottomStates, PageStates } from "../constants/States";
+import temporaryDB from '../../temporaryDB.json';
+
+const initialState: State = {
+    headerState: HeaderStates.title,
+    bottomState: BottomStates.itemSelection,
+    pageState: PageStates.itemList,
+    store: {
+        searchQuery: "",
+        suggestions: temporaryDB['suggestions'], // FIXME: should be empty at start (currently testing)
+        items: temporaryDB['testItems'] // FIXME: should be empty at start (currently testing)
+    }
+};
+
+// combining reducers for action and state
+export default function rootReducer(state: State = initialState, action: Action) {
+    const reducersToUse = [setActions, setState];
+
+    let newState = undefined;
+    for (let i in reducersToUse) {
+        newState = reducersToUse[i](state, action);
+        if (newState) { // return state if the state and action are found in the reducersToUse
+            return newState;
+        }
+    }
+
+    return state;
+}
```

Comment:
I read a bit on reducers and actions and want to make sure I understand at a high level what is happening. When something needs to be changed, I'd pass an action to dispatch(). Then the reducer would receive the action(s) and update the state accordingly? Sorry if this is basic but I have never used redux before.

Reply:
Yup, that is how it works!
(Also no worries. Also the first time I'm working with Redux as well)